client-js/app/elements/labrad-nodes.ts

```diff
@@ -408,15 +421,31 @@ export class LabradNodes extends polymer.Base {
       }
     }
 
-    this.removeItemFromList(msg);
+    this.removeNode(msg);
   }
 
 
-  onServerStatus(msg: ServerStatusMessage): void {
+  // Called when a server instance changes state on a particular node.
+  //
+  // We broadcast the status change to other instances and also update the node
+  // status in case this change needs to be reflected in the UI.
+  async onServerStatus(msg: ServerStatusMessage): Promise<void> {
+    this.broadcastStatusChange(msg);
+    try {
+      const node = await this.api.getNodeStatus(msg.node);
+      this.updateNode(node);
+    } catch (error) {
+      console.warn(`Failed to update status for ${msg.node}`, error);
+    }
+  }
+
+
+  // Broadcast status change to other instance controllers of the same server.
+  async broadcastStatusChange(msg: ServerStatusMessage): Promise<void> {
     const instances = Polymer.dom(this.root)
                              .querySelectorAll('labrad-instance-controller');
     for (const inst of instances) {
-      const instance = <any>inst;
+      const instance = inst as LabradInstanceController;
```

Comment:
In this case the object we get from the DOM actually _is_ a polymer component, the typescript compiler just doesn't know that because we got it through the untyped `querySelectorAll` interface. We need a cast to convince the compiler that this has a `.name` attribute, and I figure casting to the actual type is better than casting to `any` :-)